packages/story-editor/src/app/story/effects/useHistoryEntry.js

```diff
@@ -103,6 +111,16 @@ function useHistoryEntry({ story, current, pages, selection, capabilities }) {
         skipAddingEntry =
           JSON.stringify(adjustedPages) === JSON.stringify(adjustedEntryPages);
       }
+
+      // skip entries that have a blob url
+      // https://github.com/GoogleForCreators/web-stories-wp/issues/10289
+      pages.map((page) => {
+        page.elements.forEach((element) => {
+          if (isBlobURL(element?.resource?.src)) {
```

Comment:
`resource.poster` can potentially also be a blob URL, would be good to account for that as well.